generative_data_prep/__main__.py

```diff
@@ -290,13 +288,36 @@ def get_categories(categories_path: str):
     return category_to_id
 
 
+def create_Logger(log_file_path: str, output_dir: str):
```

Comment:
I don't think you need this function, this should all probably be called inside the Logger `__new__`

Reply:
sorry take this back, I think the better way would be to have a method inside the logger class to update the log file path, everything else should happen inside Logger `__new__`